From ff3f00b7f33332ebf1c3c05abc4a781684775b3c Mon Sep 17 00:00:00 2001
From: Maxim Cournoyer <maxim.cournoyer@gmail.com>
Date: Tue, 14 Jul 2020 11:50:12 -0400
Subject: [PATCH] config: Drop rubocop-performance, rubocop-rspec requirements.

This patch removes Rubocop extensions from Rubocop's closure to break
a dependency cycle with itself.

* .rubocop.yml: Remove rubocop-performance and rubocop-rspec and their
corresponding directives.
* .rubocop_todo.yml: Likewise.
---
 .rubocop.yml      | 15 ---------------
 .rubocop_todo.yml | 44 --------------------------------------------
 2 files changed, 59 deletions(-)

diff --git a/.rubocop.yml b/.rubocop.yml
index 4f05d5be2..f12ef7c06 100644
--- a/.rubocop.yml
+++ b/.rubocop.yml
@@ -3,8 +3,6 @@
 inherit_from: .rubocop_todo.yml
 require:
   - rubocop/cop/internal_affairs
-  - rubocop-performance
-  - rubocop-rspec
 
 AllCops:
   NewCops: enable
@@ -106,16 +104,3 @@ Metrics/ClassLength:
 Metrics/ModuleLength:
   Exclude:
     - 'spec/**/*.rb'
-
-RSpec/FilePath:
-  Exclude:
-    - spec/rubocop/formatter/junit_formatter_spec.rb
-
-RSpec/PredicateMatcher:
-  EnforcedStyle: explicit
-
-RSpec/MessageSpies:
-  EnforcedStyle: receive
-
-RSpec/NestedGroups:
-  Max: 7
diff --git a/.rubocop_todo.yml b/.rubocop_todo.yml
index 3f72042d1..fa637cd42 100644
--- a/.rubocop_todo.yml
+++ b/.rubocop_todo.yml
@@ -24,47 +24,3 @@ Metrics/MethodLength:
 # Configuration parameters: CountComments.
 Metrics/ModuleLength:
   Max: 132
-
-# Offense count: 10
-RSpec/AnyInstance:
-  Exclude:
-    - 'spec/rubocop/cli_spec.rb'
-    - 'spec/rubocop/cop/lint/duplicate_methods_spec.rb'
-    - 'spec/rubocop/cop/team_spec.rb'
-    - 'spec/rubocop/target_finder_spec.rb'
-
-# Offense count: 981
-# Configuration parameters: Prefixes.
-# Prefixes: when, with, without
-RSpec/ContextWording:
-  Enabled: false
-
-# Offense count: 3810
-# Configuration parameters: Max.
-RSpec/ExampleLength:
-  Enabled: false
-
-# Offense count: 38
-RSpec/ExpectOutput:
-  Exclude:
-    - 'spec/rubocop/cli/cli_auto_gen_config_spec.rb'
-    - 'spec/rubocop/cli/cli_options_spec.rb'
-    - 'spec/rubocop/config_spec.rb'
-    - 'spec/rubocop/cop/cop_spec.rb'
-    - 'spec/rubocop/formatter/disabled_config_formatter_spec.rb'
-    - 'spec/rubocop/formatter/formatter_set_spec.rb'
-    - 'spec/rubocop/options_spec.rb'
-    - 'spec/rubocop/rake_task_spec.rb'
-    - 'spec/rubocop/result_cache_spec.rb'
-    - 'spec/rubocop/target_finder_spec.rb'
-
-# Offense count: 434
-RSpec/MultipleExpectations:
-  Max: 25
-
-# Offense count: 5
-RSpec/SubjectStub:
-  Exclude:
-    - 'spec/rubocop/config_spec.rb'
-    - 'spec/rubocop/formatter/json_formatter_spec.rb'
-    - 'spec/rubocop/formatter/progress_formatter_spec.rb'
-- 
2.27.0

